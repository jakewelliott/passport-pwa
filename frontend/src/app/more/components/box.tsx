import { cn } from '@/lib/cn-helper';

interface BoxProps {
    title: string;
    headerClass: string;
    bodyClass: string;
    children: React.ReactNode;
}

export const Box = ({ title, headerClass, bodyClass, children }: BoxProps) => {
    return (
        <div className='flex-col'>
            <div
                className={cn(
                    'flex w-full items-center justify-center p-2 pt-3 text-center text-system_white',
                    headerClass,
                )}
            >
                <h3>{title}</h3>
            </div>
            <div className={cn('p-4', bodyClass)}>{children}</div>
        </div>
    );
};
