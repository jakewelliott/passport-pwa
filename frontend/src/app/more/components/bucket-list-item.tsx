import { useParks } from '@/hooks/queries/useParks';
import { FaRegCheckSquare, FaRegSquare } from 'react-icons/fa';
import DateHelper from '@/lib/date-helper';
import { useLocation } from '@/hooks/useLocation';
import { useUpdateBucketListItem } from '@/hooks/queries/useBucketList';
import { CompletedBucketListItem, BucketListItemInterface } from '@/lib/mock/types';

interface BucketListItemProps {
  bucketListItem: BucketListItemInterface;
  completedItems: CompletedBucketListItem[];
}

export const BucketListItem = ({ bucketListItem, completedItems }: BucketListItemProps) => {
  const isCompleted = completedItems.some(item => item.id === bucketListItem.id);
  const { data: parks } = useParks();
  const park = parks?.find(park => park.id === bucketListItem.park);
  const useUpdateBucketListItemMutation = useUpdateBucketListItem();
  const userLocation = useLocation();

  function updateBucketListItem(id: number, status: string): void {
    const useUpdateBucketListItemMutation = useUpdateBucketListItem(id);
    useUpdateBucketListItemMutation.mutate({
      latitude: userLocation.geopoint?.latitude ?? 0, 
      longitude: userLocation.geopoint?.longitude ?? 0, 
      status: status, 
      updatedAt: new Date()
    }, {
      onSuccess: () => {
        if (status === 'completed') {
          completedItems.push({
            id: bucketListItem.id,
            park: bucketListItem.park,
            updatedAt: DateHelper.stringify(new Date())
          });
        } else {
          completedItems = completedItems.filter(item => item.id !== bucketListItem.id);
        }
      }
    });
  }

  return (
    <div className='my-2.5 flex items-start'>
      {isCompleted ? (
        <FaRegCheckSquare
          data-testid='checked-icon'
          size={'24px'}
          strokeWidth={3}
          style={{ paddingRight: '5px', paddingTop: '3px' }}
          onClick={() => updateBucketListItem(bucketListItem.id, 'not completed')}
        />
      ) : (
        <FaRegSquare
          data-testid='unchecked-icon'
          size={'24px'}
          strokeWidth={3}
          style={{ paddingRight: '5px', paddingTop: '3px' }}
          onClick={() => updateBucketListItem(bucketListItem.id, 'completed')}
        />
      )}
      <div className='flex w-full flex-col justify-center'>
        <p>{bucketListItem.task}</p>
        <div className='flex flex-wrap items-center justify-between gap-2'>
          <p className='p-mini text-main_green'>{park?.addresses[0].city ?? ''}</p>
          {isCompleted && (
            <p className='p-mini'>{DateHelper.toStringShort(new Date(completedItems.find(item => item.id === bucketListItem.id)?.updatedAt ?? ''))}</p>
          )}
        </div>
      </div>
    </div>
  );
};
