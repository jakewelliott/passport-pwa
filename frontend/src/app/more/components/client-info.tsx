import DateHelper from '@/lib/date-helper';

export const ClientInfo = () => {
    // TODO: there's definitely a better way to do this
    const appVersion = '1.0.0';
    // TODO: this doesn't work for some reason
    const connectionStatus = window.navigator.onLine ? 'Online' : 'Offline';
    // TODO: implement this
    const lastCached = DateHelper.toStringLong(new Date());

    return (
        <div className='grid w-full grid-cols-2'>
            <p>App Version:</p>
            <p className='text-right'>{appVersion}</p>
            <p>Connection Status:</p>
            <p className='text-right'>{connectionStatus}</p>
            <p>Last Cached:</p>
            <p className='text-right'>{lastCached}</p>
        </div>
    );
};
