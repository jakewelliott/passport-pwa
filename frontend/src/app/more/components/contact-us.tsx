import { Link } from 'react-router-dom';
import { FaFacebookF, FaYoutube, FaInstagram } from 'react-icons/fa';

export const ContactUs = () => {
  return (
    <div className='p-4'>
      <div className='bg-supporting_inactiveblue p-2 text-center text-system_white'>
        <h3>CONTACT US</h3>
      </div>
      <div className='space-y-2 bg-supporting_lightblue p-2.5 text-center'>
        <div>
          <Link to={'www.ncparks.gov'} className='text-supporting_inactiveblue no-underline'>
            www.ncparks.gov
          </Link>
        </div>
        <div className='flex justify-center gap-3'>
          <Link
            to={'https://www.facebook.com/NorthCarolinaStateParks/'}
            className='flex h-8 w-8 items-center justify-center rounded-full bg-supporting_inactiveblue'
          >
            <FaFacebookF size={20} className='text-supporting_lightblue' />
          </Link>
          <Link
            to={'https://www.instagram.com/ncstateparks/'}
            className='flex h-8 w-8 items-center justify-center rounded-full bg-supporting_inactiveblue'
          >
            <FaInstagram size={20} className='text-supporting_lightblue' />
          </Link>
          <Link
            to={'https://www.youtube.com/@parksNC'}
            className='flex h-8 w-8 items-center justify-center rounded-full bg-supporting_inactiveblue'
          >
            <FaYoutube size={20} className='text-supporting_lightblue' />
          </Link>
        </div>
        <p className='space-y-1'>
          <span>NC Division of Parks and Recreation</span>
          <br />
          <span>Dept. of Natural and Cultural Resources</span>
          <br />
          <span>1615 Mail Service Center</span>
          <br />
          <span>Raleigh, NC 27699</span>
          <br />
          <span>(919) 707-9300</span>
        </p>
      </div>
    </div>
  );
};
