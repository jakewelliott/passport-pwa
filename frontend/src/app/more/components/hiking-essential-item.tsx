import type React from 'react';

interface HikingEssentialItemProps {
    imageSrc: string;
    altText: string;
    title: string;
    description: string;
}
