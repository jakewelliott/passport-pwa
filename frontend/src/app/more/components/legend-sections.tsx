import { BLAZE_ICONS, BLUE_ICONS, GREEN_ICONS, type PARK_ICONS, type ParkIcon, RED_ICONS } from '@/types/icons';

type IconWithExtraText = ParkIcon;

interface IconSection {
    sectionName: string;
    icons: IconWithExtraText[];
}

type IconName = (typeof PARK_ICONS)[number]['iconName'];

const textHelper = (icon: ParkIcon) => ({
    ...icon,
});

// @biome-ignore lint/style/noNonNullAssertion: we know this icon exists
const ICON_SECTIONS: IconSection[] = [
    {
        sectionName: 'Camping',
        icons: GREEN_ICONS.map(textHelper),
    },
    {
        sectionName: 'Activities',
        icons: RED_ICONS.map(textHelper),
    },
    {
        sectionName: 'Amenities',
        icons: BLUE_ICONS.map(textHelper),
    },
    {
        sectionName: 'State Trail Stamps',
        icons: BLAZE_ICONS.map(textHelper),
    },
];

const path = (iconName: IconName) => `/icons/park/${iconName}.svg`;

const IconView = ({ icon }: { icon: IconSection['icons'][number] }) => {
    return (
        <div key={icon.iconName} className='my-1 flex items-center'>
            <img src={path(icon.iconName)} alt={icon.tooltip} width={'36px'} height={'36px'} />
            <div className='ml-2 flex w-full flex-col justify-center'>
                <p>{icon.tooltip}</p>
                {icon.extraText && <p className='max-w-56 p-mini'>*{icon.extraText}</p>}
            </div>
        </div>
    );
};

export const LegendSections = () => {
    return (
        <>
            {ICON_SECTIONS.map((section) => (
                <div className='m-4' key={section.sectionName}>
                    <h4 className={`pb-3 text-icon_${section.sectionName.toLowerCase()}`}>{section.sectionName}</h4>
                    {section.icons.map((icon) => (
                        <IconView key={icon.iconName} icon={icon} />
                    ))}
                </div>
            ))}
        </>
    );
};
