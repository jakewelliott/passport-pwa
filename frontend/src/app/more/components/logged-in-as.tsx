import { useUser } from '@/hooks/queries/useUser';
import { useNavigate } from 'react-router-dom';

export const LoggedInAs = () => {
	const { data: user } = useUser();
	const navigate = useNavigate();


	const message = user?.username ? `You are currently logged in as '${user.username}'` : 'You are not logged in';

	return (
		<div className='text-center'>
			{message}
			<br />
			<button className='link' onClick={() => navigate('/logout')} type='button'>
				Log out
			</button>
		</div>
	);
};
