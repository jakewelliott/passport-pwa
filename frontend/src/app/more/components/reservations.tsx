import { Link } from 'react-router-dom';

export const Reservations = () => {
  return (
    <div className='p-4'>
      <div className='mb-3 flex items-center'>
        <img src='/icons/misc/Camping-Black.svg' alt='' className='mr-2 h-[50px] w-[50px]' />
        <div className='flex w-full flex-col justify-center'>
          <h4 className='pb-1 text-center text-main_green'>MAKE A RESERVATION</h4>
          <div className='flex flex-wrap items-center justify-between gap-2'>
            <Link to={'https://www.ncparks.gov/reservations'} className='p-mini'>
              ncparks.gov/reservations
            </Link>
            <p className='p-mini'>1-877-722-6762</p>
          </div>
        </div>
      </div>
      <div className='mt-3 flex items-center'>
        <img src='/icons/misc/PicnicShelter-Black.svg' alt='' className='mr-2 h-[50px] w-[50px]' />
        <p className='p-mini'>Reserve campsites, picnic shelters and other park facilities online or over the phone.</p>
      </div>
    </div>
  );
};
