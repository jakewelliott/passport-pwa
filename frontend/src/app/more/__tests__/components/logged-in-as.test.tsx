import { useUser } from '@/hooks/queries/useUser';
import { setupTestEnv } from '@/lib/testing/test-wrapper';
import { fireEvent, screen } from '@testing-library/react';
import { beforeAll, describe, expect, it } from 'vitest';
import { LoggedInAs } from '../../components/logged-in-as';

const { render, checkHook } = setupTestEnv();
describe('LoggedInAs', () => {
    beforeAll(async () => {
        await checkHook(useUser, 'useUser');
    });

    it('matches snapshot', () => {
        const { container } = render(<LoggedInAs />);
        expect(container).toMatchSnapshot();
    });

    it('calls logout function when logout button is clicked', () => {
        render(<LoggedInAs />);
        const logoutLink = screen.getByText('Log out');
        fireEvent.click(logoutLink);
        // TODO: find some way to test that the logout function is called
        expect(false).toBe(true);
    });
});
