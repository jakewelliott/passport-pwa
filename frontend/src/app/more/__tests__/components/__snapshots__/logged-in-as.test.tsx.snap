// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoggedInAs > matches snapshot 1`] = `
<div>
  <div
    class="text-center"
  >
    You are currently logged in as 'superAdmin'
    <br />
    <button
      class="link"
      type="button"
    >
      Log out
    </button>
  </div>
</div>
`;
